R/checks.R

```diff
@@ -1,101 +1,107 @@
+    cli::cli_abort("{.arg {input}} should be {an} numeric vector.")
   }
   invisible(NULL)
 }
 
-check_categorical <- function(x) {
+check_categorical <- function(x, call) {
   if (!is.character(x) & !is.factor(x)) {
-    rlang::abort("`x` should be a character or factor vector.")
+    cli::cli_abort("{.arg x} should be a character or factor vector.")
   }
   invisible(NULL)
 }
 
-check_unit_range <- function(x) {
-  msg <- "Desirability values should be numeric and complete in the range [0, 1]."
-  if (!is.vector(x) || !is.numeric(x)) {
-    rlang::abort(msg)
-  }
+out_of_unit_range <- function(x) {
   x <- x[!is.na(x)]
-  if (length(x) > 0 && any(x < 0 | x > 1)) {
-    rlang::abort(msg)
-  }
-  invisible(NULL)
+  any(x < 0 | x > 1)
 }
 
-check_value_order <- function(low, high, target = NULL) {
-  if (length(low) != 1 || !is.numeric(low) || is.na(low)) {
-    rlang::abort("'low' should be a single numeric value.")
+check_unit_range <- function(x, call) {
+  msg <- c(
+    "Desirability values should be numeric and complete in the range [0, 1].",
+    "i" = "Current values are {.obj_type_friendly {x}}.",
+    "i" = "{?Some v/V}alue{?s} {?are/is} outside the [0, 1] range."
+  )
+
+  if (!is.vector(x) || !is.numeric(x)) {
+    cli::cli_abort(msg, call = call)
   }
 
-  if (length(high) != 1 || !is.numeric(high) || is.na(high)) {
-    rlang::abort("'high' should be a single numeric value.")
+  if (out_of_unit_range(x) | length(x) > 1) {
+    cli::cli_abort(msg, call = call)
   }
 
+  invisible(NULL)
+}
+
+check_value_order <- function(low, high, target = NULL, call) {
+  check_number_decimal(low, call = call)
+  check_number_decimal(high, call = call)
+  check_number_decimal(target, allow_null = TRUE, call = call)
+
   if (!is.null(target)) {
-    if (length(target) != 1 || !is.numeric(target) || is.na(target)) {
-      rlang::abort("'target' should be a single numeric value.")
-    }
     ord <- low < target & target < high
     if (!ord) {
-      rlang::abort("The values should be `low < target < high`.")
-    }
-  } else {
-    ord <- low < high
-    if (!ord) {
-      rlang::abort("The values should be `low < high`.")
+      cli::cli_abort("The values should be {.code low < target < high}  (actual
+                     are {low}, {target}, and {high}).", call = call)
     }
   }
+
+  ord <- low < high
+  if (!ord) {
+    cli::cli_abort("The values should be {.code low < high} (actual are {low}
+                   and {high}).", call = call)
+  }
+
   invisible(NULL)
 }
 
-is_vector_args <- function(values, d) {
+is_vector_args <- function(values, d, call) {
   if (!is.vector(values) || !is.numeric(values)) {
-    rlang::abort("'values' should be a numeric vector.")
+    cli::cli_abort("{.arg values} should be a numeric vector.", call = call)
   }
   if (!is.vector(d) || !is.numeric(d)) {
-    rlang::abort("'d' should be a numeric vector.")
+    cli::cli_abort("'d' should be a numeric vector.", call = call)
   }
```

Comment:
could be replaced with `check_number_decimal()`

Reply:
Same as above; it requires vector inputs